step-templates/azure-database-execute-sql-cmd.json

```diff
@@ -0,0 +1,135 @@
+        }
+      },
+      {
+        "Id": "12590219-613f-4556-908a-cd87c509449f",
+        "Name": "azDbSqlCmd.ServerName",
+        "Label": "SQL Server Name",
+        "HelpText": "The name of the sql server. The FQDN (`database.windows.net`) will automatically be appended to this when needed.",
+        "DefaultValue": "",
+        "DisplaySettings": {
+          "Octopus.ControlType": "SingleLineText"
+        }
+      },
+      {
+        "Id": "3fe35f76-f140-48df-be0c-2a927069bc8b",
+        "Name": "azDbSqlCmd.dbName",
+        "Label": "Database Name",
+        "HelpText": "The name of the database to execute the sql command against",
+        "DefaultValue": "",
+        "DisplaySettings": {
+          "Octopus.ControlType": "SingleLineText"
+        }
+      },
+      {
+        "Id": "bbe9a203-89db-4286-a684-c95748543805",
+        "Name": "azDbSqlCmd.AuthType",
+        "Label": "SQL Server Authentication Type",
+        "HelpText": "The type of authentication to use when connecting.",
+        "DefaultValue": "sql",
+        "DisplaySettings": {
+          "Octopus.ControlType": "Select",
+          "Octopus.SelectOptions": "sql|SQL Authentication\nadPwd|Active Directory Password\nad|Active Directory Integration"
+        }
+      },
+      {
+        "Id": "539885ef-8ff9-407e-8e33-3bf211d58df6",
+        "Name": "azDbSqlCmd.userId",
+        "Label": "User Id",
+        "HelpText": "The user id to use when authenticating with the sql server",
+        "DefaultValue": "",
+        "DisplaySettings": {
+          "Octopus.ControlType": "SingleLineText"
+        }
+      },
+      {
+        "Id": "60af2045-ec59-4f4f-baf4-79e0ae8bd60d",
+        "Name": "azDbSqlCmd.userPwd",
+        "Label": "User Password",
+        "HelpText": "Used with SQL or Active directory Password authentication.",
+        "DefaultValue": "",
+        "DisplaySettings": {
+          "Octopus.ControlType": "Sensitive"
+        }
+      },
+      {
+        "Id": "e631537b-c109-4678-a7e5-bbc8166cca78",
+        "Name": "azDbSqlCmd.connectionTimeout",
+        "Label": "Connection Timeout",
+        "HelpText": "The timeout for the query measured in seconds between 0 and 65534",
+        "DefaultValue": "0",
+        "DisplaySettings": {
+          "Octopus.ControlType": "SingleLineText"
+        }
+      },
+      {
+        "Id": "9f787938-012a-45e9-ae58-5a06b1f4f012",
+        "Name": "azDbSqlCmd.sqlCmd",
+        "Label": "SQL Command",
+        "HelpText": "The sql command to execute",
+        "DefaultValue": "",
+        "DisplaySettings": {
+          "Octopus.ControlType": "MultiLineText"
+        }
+      },
+      {
+        "Id": "273c678e-008e-4598-978e-9d7abaadbf6c",
+        "Name": "azDbSqlCmd.retainResults",
+        "Label": "Retain Results",
+        "HelpText": "The results file is saved for future analysis and readability",
+        "DefaultValue": "false",
+        "DisplaySettings": {
+          "Octopus.ControlType": "Checkbox"
```

Comment:
Chris, 
As we chatted about, I don't think saving to a file is the best option here as it assumes that the following steps will execute on the same machine (not necessarily true). 

I think we should replace this parameter with two parameters: 

1. something like "Output to variable", and if set save the CSV to an Octopus output variable: https://octopus.com/docs/projects/variables/output-variables
2. something like "Octopus artifact name" which if set will create an artifact: https://octopus.com/docs/deployment-process/artifacts

That way the user can control whether they want neither, one, or both of an output variable and an artifact.

Reply:
Addressed in commit 305ce34